tests.py

```diff
@@ -176,3 +205,66 @@ def test_amend_requirements_content_replacement_amonst_others_2(self):
             requirements, 'autocompeter', new_lines
         )
         self.assertEqual(result, previous + new_lines)
+
+    @httpretty.activate
+    @cleanup_tmpdir('peepin*')
+    def test_run(self):
+        httpretty.register_uri(
+            httpretty.GET,
+            "https://pypi.python.org/pypi/peepin",
+            body="""
+            <div id="content">
+
+            <div id="breadcrumb">
+              <a href="/pypi">Package Index</a>
+
+                <span class="breadcrumb-separator">&gt;</span>
+                <a href="/pypi/peepin">peepin</a>
+
+                <span class="breadcrumb-separator">&gt;</span>
+                <a href="/pypi/peepin/0.3">0.3</a>
+            </div>
+            """,
+        )
+        httpretty.register_uri(
+            httpretty.GET,
+            "https://pypi.python.org/pypi/peepin/0.10",
+            # sorry
+            body="""
+<a href="https://pypi.python.org/packages/2.7/p/peepin/peepin-0.10-py2-none-any.whl#md5=0a"
+    >peepin-0.10-py2-none-any.whl</a>
+
+<a href="https://pypi.python.org/packages/3.3/p/peepin/peepin-0.10-py3-none-any.whl#md5=45"
+    >peepin-0.10-py3-none-any.whl</a>
+
+<a href="https://pypi.python.org/packages/source/p/peepin/peepin-0.10.tar.gz#md5=ae"
+    >peepin-0.10.tar.gz</a>
+            """,
+        )
+        httpretty.register_uri(
+            httpretty.GET,
+            "https://pypi.python.org/packages/2.7/p/peepin/peepin-0.10-py2-none-any.whl#md5=0a",
+            body="Some py2 wheel content\n",
+        )
+        httpretty.register_uri(
+            httpretty.GET,
+            "https://pypi.python.org/packages/3.3/p/peepin/peepin-0.10-py3-none-any.whl#md5=45",
+            body="Some py3 wheel content\n",
+        )
+        httpretty.register_uri(
+            httpretty.GET,
+            "https://pypi.python.org/packages/source/p/peepin/peepin-0.10.tar.gz#md5=ae",
+            body="Some tarball content\n",
+        )
+        with tmpfile() as filename:
+            open(filename, 'w').write('')
+            retcode = peepin.run('peepin==0.10', filename)
+            self.assertEqual(retcode, 0)
+            output = open(filename).read()
```

Comment:
Not strictly necessary, but it would be good hygiene to close this file handle as well.